api/preprints/permissions.py

```diff
@@ -138,3 +138,18 @@ def has_object_permission(self, request, view, obj):
                 raise exceptions.PermissionDenied(detail='Withdrawn preprints may not be edited')
             return True
         raise exceptions.NotFound
+
+
+class PreprintInstitutionPermission(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if obj.is_public:
+            return True
+
+        auth = get_user_auth(request)
+        if not auth.user:
+            return False
+
+        if request.method in permissions.SAFE_METHODS:
+            return obj.has_permission(auth.user, 'read')
+        else:
+            return obj.has_permission(auth.user, 'write')
```

Comment:
I don't think this is correct if you look at how `can_edit` is implemented:
```
    def can_edit(self, auth=None, user=None):
        """Return if a user is authorized to edit this preprint.
        Must specify one of (`auth`, `user`).

        :param Auth auth: Auth object to check
        :param User user: User object to check
        :returns: Whether user has permission to edit this node.
        """
        if not auth and not user:
            raise ValueError('Must pass either `auth` or `user`')
        if auth and user:
            raise ValueError('Cannot pass both `auth` and `user`')
        user = user or auth.user

        return (
            user and ((self.has_permission(user, WRITE) and self.has_submitted_preprint) or self.has_permission(user, ADMIN))
        )
 ```
 All it does additionally is check the `has_submitted_preprint` is True, which is not the behavior product wants. I can't change `can_edit` without checking going through the repo and re-writing the permissions to make that check, because for some actions it's necessary to confirm the `has_submitted_preprint` is True. 

 Similarly this is the permissions for `can_view`:
 ```
     def can_view(self, auth):
        if not auth.user:
            return self.verified_publishable

        return (self.verified_publishable or
            (self.is_public and auth.user.has_perm('view_submissions', self.provider)) or
            self.has_permission(auth.user, ADMIN) or
            (self.is_contributor(auth.user) and self.has_submitted_preprint)
        )
```
It has a lot of random checks that don't relate to Product requirements.

Admittedly this permission is incorrect because it checks for unsafe methods, which is not something the view allows, so I've removed those lines and added an explicit check forbidding that within the permission class to make sure of that nobody adds it to a POST request in the the future.

Reply:
With WRITE contributor can edit, the `can_edit()` no longer needs this extra check.

```python
self.has_permission(user, WRITE) and self.has_submitted_preprint
```

It was there to make sure submitter can still edit the preprint if they are downgraded from ADMIN to WRITE, which is no longer relevant.

For now, all we need for `can_edit()` is.

```python
return user and self.has_permission(user, WRITE) 
```

That being said, I agree we should keep the `PreprintInstitutionPermission` not using the `can_view()` or `can_edit()`.